src/swipes.py

```diff
@@ -0,0 +1,304 @@
+              'end_time': '',
+              'session_type': '',
+              'location': '',
+              'start_time': '',
+              'swipes': 0,
+              'weekday': '',
+              BRB_ONLY: False,
+              DINING_HALL: False,
+              TRILLIUM: False,
+          }, index=[0])
+          print('new marker made')
+        if 'end' in update_info and update_info['end'].iloc[0]['date'] == timestamp_str:
+          print('hit time marker from last update on {}'.format(timestamp_str))
+          break
+        try:
+          timestamp = datetime.strptime(timestamp_str, '%Y-%m-%d %I:%M:00 %p')
+        except:
+          try:
+            timestamp = datetime.strptime(timestamp_str, '%Y-%m-%d %I:%M:01 %p')
+          except:
+            continue
+        date = timestamp.date()
+        # print(timestamp_str)
+        session_type = sort_session_type(date, breaks)
+        if session_type != 'regular':
+          print(session_type)
+        weekday = weekdays[timestamp.weekday()]
+        # sort time into a time block
+        if timestamp.minute > 30:
+          delta = timedelta(minutes=30)
+          start_time = timestamp.strftime('%I:30 %p')
+          end_time = (timestamp + delta).strftime('%I:00 %p')
+        elif timestamp.minute == 0:
+          delta = timedelta(minutes=1)
+          start_time = (timestamp - delta).strftime('%I:30 %p')
+          end_time = timestamp.strftime('%I:00 %p')
+        elif timestamp.minute <= 30:
+          start_time = timestamp.strftime('%I:00 %p')
+          end_time = timestamp.strftime('%I:30 %p')
+
+        for place in obj['UNITS']:
+          location = place['UNIT_NAME']
+          # remove locations that are not eateries we care about
+          if location not in LOCATION_NAMES:
+            continue
+          data_list.append(pd.DataFrame(data={
+              'date': timestamp.strftime('%m/%d/%Y'),
+              'end_time': end_time,
+              'session_type': session_type,
+              'location': location,
+              'start_time': start_time,
+              'swipes': place['CROWD_COUNT'],
+              'weekday': weekday,
+              BRB_ONLY: True if LOCATION_NAMES[location]['type'] == BRB_ONLY else False,
+              DINING_HALL: True if LOCATION_NAMES[location]['type'] == DINING_HALL else False,
+              TRILLIUM: True if LOCATION_NAMES[location]['type'] == TRILLIUM else False,
+              }, index=[0]))
+        i += 1
+        line_counter += 1
+
+      print('done parsing data.log')
+      if 'old_data' in update_info:
+        data_list.append(update_info['old_data'])
+      df = pd.concat(data_list)
+      df.to_csv(MAIN_CSV, header=True, index=False)
+      data_file = sort_by_timeblock(MAIN_CSV)
+      # add new marker to front of table to be used next time for time tracking
+      df = pd.concat([update_info['new_marker'], df])
+      df.to_csv(MAIN_CSV, header=True, index=False)
+      return data_file
```

Comment:
there needs to be more whitespace in the above function so it's easier to follow, add newlines between different sections of code!